### PR TITLE
Accurately consider inactive displays when signaling child

### LINKED ISSUE #3242: i3status/blocks bar hangs up from time to time
From time to time i3status/blocks bar hangs up.
I think the bar hangs often after suspend/resume. If CPU usage reaches 100% the bar hangs up, too.

Example: Now is 10:57, but the bar did hang at 10:39:

PS:
Binary i3 version:  4.13 (2016-11-08) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.13 (2016-11-08) (pid 5895)o abort…)
Loaded i3 config: /home/unera/.i3/config (Last modified: Ср 07 мар 2018 21:56:48, 3329319 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3

https://logs.i3wm.org/logs/5703935746637824.bz2

### PR DESCRIPTION
**Pull request by dfoxfranke**
This should fix #3242. Marked as draft because I haven't tested this yet.

### CODE REVIEW

File: i3bar/src/outputs.c [hunk 1]
```diff
@@ -46,7 +46,11 @@ static int outputs_boolean_cb(void *params_, int val) {
     struct outputs_json_params *params = (struct outputs_json_params *)params_;
 
     if (!strcmp(params->cur_key, "active")) {
-        params->outputs_walk->active = val;
+        bool changed = params->outputs_walk->active != (bool)val;
+        if (changed) {
+            params->outputs_walk->active = val;
+            stop_or_cont_child();
```
Comment (stapelberg) [inline i3bar/src/outputs.c]: Can you move this stop_or_cont_child call into i3bar/src/ipc.c:got_output_reply next to the reconfig_windows calls please? I think it’d be cleaner to modify state *after* parsing, not *while* parsing :)

File: i3bar/src/outputs.c [hunk 2]
```diff
@@ -46,7 +46,11 @@ static int outputs_boolean_cb(void *params_, int val) {
     struct outputs_json_params *params = (struct outputs_json_params *)params_;
 
     if (!strcmp(params->cur_key, "active")) {
-        params->outputs_walk->active = val;
+        bool changed = params->outputs_walk->active != (bool)val;
```
Comment (stapelberg) [inline i3bar/src/outputs.c]: nit: here and below, please:

1. Mark these bools as const.
2. Add parens around the expression.

File: i3bar/include/child.h [hunk 3]
```diff
@@ -75,6 +75,13 @@ void stop_child(void);
  */
 void cont_child(void);
 
+/*
```
Comment (stapelberg) [inline i3bar/include/child.h]: nit: use `/**` so that doxygen will pick up the comment.

**Comment by Airblader** [comment]
@dfoxfranke Are you planning on following up on this? :-)

**Pull request closed without merging**